src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -51,711 +77,257 @@ public class AnalysisService : IDisposable
-            ActiveRules = activeRules.ToArray();
+            _loggerFactory = loggerFactory;
+            _logger = loggerFactory.CreateLogger<AnalysisService>();
             _languageServer = languageServer;
             _configurationService = configurationService;
-            _logger = logger;
-            _pssaModuleInfo = pssaModuleInfo;
-            _mostRecentCorrectionsByFile = new ConcurrentDictionary<string, (SemaphoreSlim, Dictionary<string, MarkerCorrection>)>();
+            _workplaceService = workspaceService;
+            _analysisDelayMillis = 750;
+            _mostRecentCorrectionsByFile = new ConcurrentDictionary<ScriptFile, CorrectionTableEntry>();
+            _analysisEngine = new Lazy<PssaCmdletAnalysisEngine>(InstantiateAnalysisEngine);
         }
 
-        #endregion // constructors
-
-        #region Public Methods
+        /// <summary>
+        /// The analysis engine to use for running script analysis.
+        /// </summary>
+        private PssaCmdletAnalysisEngine AnalysisEngine => _analysisEngine.Value;
 
         /// <summary>
-        /// Factory method for producing AnalysisService instances. Handles loading of the PSScriptAnalyzer module
-        /// and runspace pool instantiation before creating the service instance.
+        /// Sets up a script analysis run, eventually returning the result.
         /// </summary>
-        /// <param name="settingsPath">Path to the PSSA settings file to be used for this service instance.</param>
-        /// <param name="logger">EditorServices logger for logging information.</param>
-        /// <returns>
-        /// A new analysis service instance with a freshly imported PSScriptAnalyzer module and runspace pool.
-        /// Returns null if problems occur. This method should never throw.
-        /// </returns>
-        public static AnalysisService Create(ConfigurationService configurationService, ILanguageServer languageServer, ILogger logger)
+        /// <param name="filesToAnalyze">The files to run script analysis on.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel this call with.</param>
+        /// <returns>A task that finishes when script diagnostics have been published.</returns>
+        public void RunScriptDiagnostics(
+            ScriptFile[] filesToAnalyze,
+            CancellationToken cancellationToken)
         {
-            Validate.IsNotNull(nameof(configurationService), configurationService);
-            string settingsPath = configurationService.CurrentSettings.ScriptAnalysis.SettingsPath;
-            try
+            if (AnalysisEngine == null)
+            {
+                return;
+            }
+
+            // Create a cancellation token source that will cancel if we do or if the caller does
+            var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+            // If there's an existing task, we want to cancel it here;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)